apps/prairielearn/src/pages/instructorAssessmentSettings/instructorAssessmentSettings.html.ts

```diff
@@ -97,47 +110,46 @@ export function InstructorAssessmentSettings({
                     id="number"
                     name="number"
                     value="${resLocals.assessment.number}"
-                    disabled
+                    ${canEdit ? '' : 'disabled'}
                   />
                   <small class="form-text text-muted">
                     The number of the assessment within the set.
                   </small>
                 </div>
                 <div class="form-group">
                   <label for="module">Module</label>
-                  <input
-                    type="text"
+                  <select
                     class="form-control"
                     id="module"
                     name="module"
-                    value="${resLocals.assessment_module
-                      ? resLocals.assessment_module.heading
-                      : ''}"
-                    disabled
-                  />
+                    ${canEdit ? '' : 'disabled'}
+                  >
+                    ${assessmentModules.map(
+                      (module) => html`
+                        <option
+                          value="${module.name}"
+                          ${resLocals.assessment_module.name === module.name ? 'selected' : ''}
+                        >
+                          ${module.name}
+                        </option>
+                      `,
+                    )}
+                  </select>
                   <small class="form-text text-muted">
                     The <a href="${resLocals.urlPrefix}/course_admin/modules">module</a> this
                     assessment belongs to.
                   </small>
                 </div>
                 <div class="form-group">
                   <label for="aid">AID</label>
-                  ${resLocals.authz_data.has_course_permission_edit &&
-                  !resLocals.course.example_course
-                    ? ChangeIdButton({
-                        label: 'AID',
-                        currentValue: resLocals.assessment.tid,
-                        otherValues: tids,
-                        csrfToken: resLocals.__csrf_token,
-                      })
-                    : ''}
                   <input
```

Comment:
Nit: I'd move this field to be the first one on the page.

The `ChangeIdButton` component used `pattern="[\\-A-Za-z0-9_\\/]+"` on the input to provide automatic client-side validation. Consider adding that here too. That would also allow you to remove the custom check for spaces that you added in recent commits.

Reply:
Moved AID field to top of the page and added input validation in [2937e62](https://github.com/PrairieLearn/PrairieLearn/pull/10677/commits/2937e62532bbb89a04b42feab9bc5e543d0b3ce0). Also, removed the custom check added in the previous commit.